src/components/shared/FormField/index.js

```diff
@@ -0,0 +1,14 @@
+import textField from './TextField';
+import fileField from './FileField';
+
+const typeToComponent = {
+  text: textField,
+  file: fileField,
+};
+
+export default {
+  functional: true,
+  // TODO do we want to inject all props from 'outputVendors' (by passing context.data),
+  // or we want to sanitize and select specific properties?
```

Comment:
The `context.data` is the object from `v-bind`. In my code, I injected the whole `fields` data from `outputVendor`.

The question is if we want to sanitize or filter some properties from the `fields` before it affects the UI.
Because now, I'm giving to `outputVendor` developer full control on the UI, because what he will define in the `fields` will become to component properties.

Reply:
As I wrote in another comment, I think we should less dynamic in the outputVendors. There won't be that many. I think they should have a clear interface for their common functionality and extentions for vendor specific behaviour and the ui will match this - common component for the common code and specific components for vendor specific behaviour.
That is what I think at the moment, I will spend some time on the output vendor api and then will be smarter